redfish-core/lib/update_service.hpp

```diff
@@ -60,6 +61,91 @@ inline static void activateImage(const std::string& objPath,
             "Active"));
 }
 
+inline void updateErrorLogMessage(
+    const std::shared_ptr<task::TaskData>& taskData,
+    const std::string& index)
+{
+    using AdditionalDataType = std::vector<std::string>;
+    using GetManagedPropertyType = boost::container::flat_map<
+        std::string,
+        std::variant<std::string, AdditionalDataType>>;
+    using GetManagedObjectsType = boost::container::flat_map<
+        sdbusplus::message::object_path,
+        boost::container::flat_map<std::string, GetManagedPropertyType>>;
+
+    crow::connections::systemBus->async_method_call(
+        [taskData, index](const boost::system::error_code errorCode,
+                    GetManagedObjectsType& resp) {
+            if (errorCode)
+            {
+                BMCWEB_LOG_ERROR
+                    << "updateErrorLogMessage returned an error "
+                    << errorCode;
+                return;
+            }
+
+            std::string* messagePtr = nullptr;
+            std::string* eventIdPtr = nullptr;
+            std::string eventId;
+            std::string message;
+            std::string addDataStr;
+
+            for (auto& objectPath : boost::adaptors::reverse(resp))
+            {
+                for (auto& interfaceMap : objectPath.second)
+                {
+                    if (interfaceMap.first == "xyz.openbmc_project.Logging."
+                                              "Entry")
+                    {
+                        for (auto& propertyMap : interfaceMap.second)
+                        {
+                            if (propertyMap.first == "Message")
+                            {
+                                messagePtr = std::get_if<std::string>(
```

Comment:
Need to check if this is nullptr

Reply:
E.g. https://github.com/ibm-openbmc/bmcweb/blob/0cb1d86c47b1b9d2fe56077ccc045571358fb183/redfish-core/lib/memory.hpp#L161
Same for the rest of these

Reply:
if the value is a nullptr, should an internalErrror be thrown? Or can that value for the task message just be blank?

Reply:
It really should not be nullptr so an internalError is fine